update/sync/concurrentTriesMap_test.go

```diff
@@ -0,0 +1,61 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/random"
+	"github.com/ElrondNetwork/elrond-go/update/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewConcurrentTriesMap(t *testing.T) {
+	t.Parallel()
+
+	ctm := newConcurrentTriesMap()
+	require.NotNil(t, ctm)
+}
+
+func TestConcurrentTriesMap_ConcurrentAccesses(t *testing.T) {
+	t.Parallel()
+
+	// when running with -race, this should not throw race conditions. If we remove the mutex protection inside the struct,
+	// this test will fail
+	testDuration := 50 * time.Millisecond
+	rnd := random.ConcurrentSafeIntRandomizer{}
+	ctx, cancel := context.WithTimeout(context.Background(), testDuration)
+	ctm := newConcurrentTriesMap()
+	go func(ctx context.Context) {
```

Comment:
maybe you can test with more than 3 goroutines

Reply:
no need for that. I created 3 goroutines which call one of the 3 methods continuously